Add shortNames aliases for ImageRepository and ImagePolicy CRDs

cappyzawa opened the pull request:
## Summary

This PR adds shortNames aliases for ImageRepository and ImagePolicy CRDs to enable shorter kubectl commands as requested in https://github.com/fluxcd/flux2/issues/5411 .

## Changes

- Add multiple aliases for ImageRepository: `imgrepo`, `imagerepo`
- Add multiple aliases for ImagePolicy: `imgpol`, `imagepol`

The aliases follow Flux ecosystem naming conventions using the `img` prefix, consistent with standard Docker/Kubernetes terminology and aligned with other Flux CRD shortNames (e.g., `gitrepo`, `helmrepo`, `hr`, `ks`). Additional aliases provide users with more flexibility in choosing their preferred shorthand.

## Verification

After applying the updated CRDs, the aliases are correctly registered:

```
$ kubectl api-resources | grep -E "imgpol|imgrepo"
imagepolicies                       imgpol,imagepol     image.toolkit.fluxcd.io/v1beta2          true         ImagePolicy
imagerepositories                   imgrepo,imagerepo   image.toolkit.fluxcd.io/v1beta2          true         ImageRepository
```

Users can now use shorter commands with multiple options:
- `kubectl get imgrepo` / `kubectl get imagerepo` instead of `kubectl get imagerepositories`
- `kubectl get imgpol` / `kubectl get imagepol` instead of `kubectl get imagepolicies`

matheuscscp commented:
Thanks very much @cappyzawa!

`imgrepo` and `imgpol` looks good, but I don't mind having a few more to be honest:

* `imagerepo` and `imagepol` (these are the two my hands type automatically every time and never work :cry:)
* `ir` and `ip`

Let's wait and see what other people think.

cappyzawa commented:
Good idea! I implemented all the additional shortNames you suggested in affdaa3. 

Added:
- ImageRepository: `imgrepo`, `imagerepo`, `ir`
- ImagePolicy: `imgpol`, `imagepol`, `ip`

I tested them locally and they all work, but heads up - `ip` conflicts with the existing endpoints shortName. You get a warning when using `kubectl get ip`, though it still works for ImagePolicy.

Feel free to give it a try!

cappyzawa commented:
Good idea! I implemented all the additional shortNames you suggested in affdaa3. 

Added:
- ImageRepository: `imgrepo`, `imagerepo`, `ir`
- ImagePolicy: `imgpol`, `imagepol`, `ip`

I tested them locally and they all work, but heads up - `ip` conflicts with the existing `ipaddresses` (networking.k8s.io/v1) shortName. You get a warning when using `kubectl get ip`, though it still works for ImagePolicy.

Feel free to give it a try!

cappyzawa commented:
Agreed! I've removed `ip` and `ir` to avoid conflicts as you suggested in 22a6b67.

matheuscscp reviewed (approved):
Thanks very much @cappyzawa! Let's wait for other people to see the PR before we merge.

stefanprodan reviewed (approved):
LGTM
